bytecomp/translcore.ml

```diff
@@ -1018,20 +1018,23 @@ and transl_exp0 e =
       | Texp_function(_, _, _)
       | Texp_construct (_, {cstr_arity = 0}, _)
         -> transl_exp e
-      | Texp_ident _
-        when not (Typeopt.lazy_val_requires_forward e.exp_env e.exp_type) ->
-          transl_exp e
-      | Texp_constant(Const_float _) | Texp_ident _ ->
+      | Texp_constant(Const_float _) ->
+          Lprim(Pmakeblock(Obj.forward_tag, Immutable, None),
```

Comment:
Please add a comment explaining that this is ok since this won't be shortcuted for floats.